kotlin/internal/toolchains.bzl

```diff
@@ -202,6 +203,15 @@ _kt_toolchain = rule(
             `kt_abi_plugin_incompatible`""",
             default = False,
         ),
+        "experimental_use_public_only_abi_jars": attr.bool(
+            doc = """Compile using public only abi jars.
+            This effectively applies the following two compiler plugin options:
+              plugin:org.jetbrains.kotlin.jvm.abi:treatInternalAsPrivate=true
```

Comment:
Do these flags need to be paired together to work?

Reply:
in my (limited) experience yes, they only appear to work together... however i will split them, maybe there are variations in compiler versions....

Reply:
maybe it would be better if we pivoted on this idea; if we keep down this road there will be a parallel release train with the kotlin compiler. Each option added for the abi plugin will be mirrored in a similar commit/release here. Could we instead have some `experimental_abi_plugin_flags` dictionary or similar that is unpacked when configuring the plugin in KotlinJvmTaskExecutor ?

On the other hand i quite like the idea of restricting what options are available such that these rules can be opinionated about how to build abi jars and increase compilation avoidance :D

Reply:
it looks like [this](https://github.com/JetBrains/kotlin/blob/9df698ee9f3693d2002b269ab944d0f5dcb75481/plugins/jvm-abi-gen/src/org/jetbrains/kotlin/jvm/abi/JvmAbiClassBuilderInterceptor.kt#L250-L255) conditional means both flags are needed as its gated behind `removePrivateClasses`...

```
private fun shouldRemoveFromAbi(irClass: IrClass?, removePrivateClasses: Boolean, treatInternalAsPrivate: Boolean): Boolean = when {
    irClass == null -> false
    irClass.isFileClass -> false
    removePrivateClasses -> irClass.isVisibilityStrippedFromAbi(stripInternal = treatInternalAsPrivate)
    else -> false
}
```

Reply:
anyway, i've pushed another commit that splits these flags... its no effort to revert that commit if we so wish